modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue

```diff
@@ -116,7 +116,9 @@ export default {
       menuItems: [],
       createMenu: [],
       patches: [],
-      editMode: window.sessionStorage.getItem('aposEditMode') === 'true'
+      editMode: window.sessionStorage.getItem('aposEditMode') === 'true',
+      idleTimer: undefined,
+      idleTriggered: false
```

Comment:
This feels like unnecessary plumbing to me. I think you should just call a method when the timeout fires.

Reply:
It took me a minute but I remembered why I did this. We can use `await` inside a `setTimeout`. There's an alternative we could do by wrapping setTimeout, but I'm not sure if we can clear it.